snapcraft/internal/db/datastore.py

```diff
@@ -0,0 +1,112 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import pathlib
+from typing import Any, Dict, List, Type
+
+import tinydb
+import yaml
+
+from . import errors, migration
+
+logger = logging.getLogger()
+
+
+class _YAMLStorage(tinydb.Storage):
+    """Provide YAML-backed storage for TinyDB."""
+
+    def __init__(self, path: str):
+        self.path = pathlib.Path(path)
+
+    def read(self) -> Dict[Any, Any]:
```

Comment:
shouldn't we have something where we always return [str, Any]?

Reply:
Hrm, this interface is really tinydb's depending on what data they send down.  It possibly could be `Any` But in all reality *should* be `Dict[str, Any]`. I'll change it to Dict[str, Any] but will validate the loaded yaml before returning.

Reply:
fixed